processor/src/decorators/mod.rs

```diff
@@ -165,14 +163,14 @@ where
     ///
     /// # Errors
     /// Returns an error if the key is already present in the advice map.
-    fn inject_mem_values(&mut self, start_addr: u32, num_words: u32) -> Result<(), ExecutionError> {
+    fn inject_mem_values(&mut self) -> Result<(), ExecutionError> {
```

Comment:
I thought about it, and IMO it is better to remove the error. Otherwise writing reliable MASM code that uses this would be kind of hard, since we can't really tell from MASM what keys are already in the map.

It seems also useful from the perspective of host communication, since the VM can update a hash that is know to both with new data.

Reply:
Agreed! This error was added at the time when we thought that the key would always be hash of the data, but we decided not to enforce such conditions since then.